lib/data_services_api/dsapi_response_converter.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module DataServicesApi
+  class DSAPIResponseConverter
+    def initialize(sapint_response, data_api)
+      @sapint_response = sapint_response
+      @data_api = data_api
+    end
+
+    # Converts SAPINT returned JSON format to DSAPI returned JSON format
+    def to_dsapi_response
+      @sapint_response['items'].map do |value|
+        to_dsapi_item(value)
+      end
+    end
+
+    private
+
+    def to_dsapi_item(sapint_item)
+      sapint_item.reduce({}) do |result, (key, value)|
+        result.merge(to_dsapi_json(key, value))
+      end
+    end
+
+    def to_dsapi_json(sapint_key, sapint_value)
+      dataset_name = @data_api[/[a-z]*$/]
+
+      return { sapint_key => sapint_value } if sapint_key == '@id'
+      return { "#{dataset_name}:#{sapint_key}" => sapint_value } unless sapint_value.is_a?(Hash)
+
+      sapint_value.map do |key, value|
+        ["#{dataset_name}:#{sapint_key}#{key == '@id' ? '' : key.capitalize}", value]
```

Comment:
When is the `key.capitalize` call used? I couldn't see any examples of this in the tests you've written

Reply:
For example if we have:
```
...
propertyAddress: {
    ...
    county: ...
    ...
}
...
```
this should become:
```
...
propertyAddressCounty: ...
...
```
I'll add a test to try to simplify what's going on for a future reader 👍